src/actions/teams.ts

```diff
@@ -138,16 +138,36 @@ export function getTeams(page = 0, perPage: number = General.TEAMS_CHUNK_SIZE, i
     };
 }
 
-export function searchTeams(term: string): ActionFunc {
-    return bindClientFunc({
-        clientFunc: Client4.searchTeams,
-        onRequest: TeamTypes.GET_TEAMS_REQUEST,
-        onSuccess: [TeamTypes.RECEIVED_TEAMS_LIST, TeamTypes.GET_TEAMS_SUCCESS],
-        onFailure: TeamTypes.GET_TEAMS_FAILURE,
-        params: [
-            term,
-        ],
-    });
+export function searchTeams(term: string, paginate = false, page = 0, perPage = General.PAGE_SIZE_DEFAULT): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        dispatch({type: TeamTypes.GET_TEAMS_REQUEST, data: null}, getState);
+
+        let response;
+        try {
+            response = await Client4.searchTeams(term, paginate, page, perPage);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(batchActions([
+                {type: TeamTypes.GET_TEAMS_FAILURE, error},
+                logError(error),
+            ]), getState);
+            return {error};
+        }
+
+        const teams = paginate ? response.teams : response;
+
+        dispatch(batchActions([
+            {
+                type: TeamTypes.RECEIVED_TEAMS_LIST,
+                data: teams,
+            },
+            {
+                type: TeamTypes.GET_TEAMS_SUCCESS,
+            },
+        ]), getState);
+
+        return {data: response};
```

Comment:
In the case that this is a non-paginated request `response` will have the same value as `teams`. For a paginated request it will look like:
```json
{"teams": [], "total_count": 0}
```

Reply:
Yeah that's how we handled it in another spot too rather than creating two separate actions.